src/dist/component/package.rs

```diff
@@ -163,19 +164,37 @@ struct MemoryBudget {
     used: usize,
 }
 
+static mut RAM_NOTICE_SHOWN: bool = false;
```

Comment:
I dislike that this will result in unsafe -- could we use some kind of interior-mutability thing instead, perhaps an `AtomicBool` -- that seems more plausible and shouldn't need unsafe.

Reply:
We can. Today we only do one unpack at a time so it is safe, but I will fix this to avoid future foot-guns.